diff --git a/include/mpv/client.h b/include/mpv/client.h
index 85cff63bd5..742729ed2a 100644
--- a/include/mpv/client.h
+++ b/include/mpv/client.h
@@ -601,6 +601,11 @@ MPV_EXPORT mpv_handle *mpv_create_weak_client(mpv_handle *ctx, const char *name)
  */
 MPV_EXPORT int mpv_load_config_file(mpv_handle *ctx, const char *filename);
 
+/**
+ Trigger an external resize event
+ */
+MPV_EXPORT void mpv_external_resize(mpv_handle *ctx);
+
 /**
  * Return the internal time in nanoseconds. This has an arbitrary start offset,
  * but will never wrap or go backwards.
diff --git a/player/client.c b/player/client.c
index e23b98b00f..c5852302fb 100644
--- a/player/client.c
+++ b/player/client.c
@@ -2140,6 +2140,11 @@ int64_t mpv_get_time_us(mpv_handle *ctx)
 
 #include "video/out/libmpv.h"
 
+void mpv_external_resize(mpv_handle *ctx)
+{
+    vo_control(ctx->mpctx->video_out, VOCTRL_EXTERNAL_RESIZE, NULL);
+}
+
 static void do_kill(void *ptr)
 {
     struct MPContext *mpctx = ptr;
